src/m365/onenote/commands/notebook/notebook-add.ts

```diff
@@ -0,0 +1,156 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { AxiosRequestConfig } from 'axios';
+import { validation } from '../../../../utils/validation';
+import { aadGroup } from '../../../../utils/aadGroup';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  name: string;
+  userId?: string;
+  userName?: string;
+  groupId?: string;
+  groupName?: string;
+  webUrl?: string;
+}
+
+class OneNoteNotebookAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.NOTEBOOK_ADD;
+  }
+
+  public get description(): string {
+    return 'Create a new OneNote notebook.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        joined: args.options.joined
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '-n, --name <name>' },
+      { option: '--userId [userId]' },
+      { option: '--userName [userName]' },
+      { option: '--groupId [groupId]' },
+      { option: '--groupName [groupName]' },
+      { option: '-u, --webUrl [webUrl]' }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId as string)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId as string)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.userId && args.options.userName) {
+          return 'Specify either userId or userName, but not both';
+        }
+
+        if (args.options.groupId && args.options.groupName) {
+          return 'Specify either groupId or groupName, but not both';
+        }
```

Comment:
We should validate those using `optionSets` to validate if one of many options is used. Also I think we should validate if we specify either user or group, right? 🤔